ui/tests/integration/components/auth/form/test-helper.js

```diff
@@ -58,4 +45,47 @@ export default (test) => {
     const [actual] = this.onSuccess.lastCall.args;
     assert.strictEqual(actual, 'success!', 'it calls onSuccess');
   });
+
+  test('it submits form data with defaults', async function (assert) {
+    await this.renderComponent();
+    const { options } = AUTH_METHOD_MAP.find((m) => m.authType === this.authType);
+    const { loginData } = options;
+
+    for (const [field, value] of Object.entries(loginData)) {
+      await fillIn(GENERAL.inputByAttr(field), value);
+    }
+    await click(AUTH_FORM.login);
+    const [actual] = this.authenticateStub.lastCall.args;
+    assert.propEqual(
+      actual.data,
+      this.expectedSubmit.default,
+      'auth service "authenticate" method is called with form data'
+    );
+  });
+
+  // not for testing real-world submit, that happens in acceptance tests
+  // component here just yields <:advancedSettings> to test form submits data from yielded inputs
+  test('it submits form data yielded inputs', async function (assert) {
+    await this.renderComponent({ yieldBlock: true });
+    const { options } = AUTH_METHOD_MAP.find((m) => m.authType === this.authType);
+    const { loginData } = options;
+
+    for (const [field, value] of Object.entries(loginData)) {
+      await fillIn(GENERAL.inputByAttr(field), value);
+    }
+
+    if (this.authType === 'token') {
```

Comment:
overall i'm trying to avoid conditionals when abstracting tests, but this felt pretty minimal. if folks don't like it I can pull out these from the helper and manually write them for each test case.